### PR TITLE
Validate username uniqueness

### PR DESCRIPTION
**Pull request by koterpillar**

### CODE REVIEW

**Comment by alex-hutton** [comment]
Looks good, ++

**Comment by alex-hutton** [comment]
Do we need tests though?

**Comment by alex-hutton** [comment]
++

File: ixprofile_client/steps.py [hunk 1]
```diff
@@ -495,6 +495,54 @@ def _user_details(self, user):
...
+
+        username = user['username']
+        exclude_email = user['email']
+
+        if any(
+            user['username'] == username
+            for email, user in self.users.items()
+            if email != exclude_email
+        ):
+            self._raise_failure({
```
Comment (alex-hutton) [inline ixprofile_client/steps.py]: Is this hash imitating the response from real profiles?

Comment (koterpillar) [inline ixprofile_client/steps.py]: Yes.

**Comment by alex-hutton** [comment]
No need for tests.

**Pull request merged**